sdk/core/azure-core/src/main/java/com/azure/core/util/ClientOptions.java

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.annotation.Fluent;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Client Options for setting common properties for example applicationId. Most of these properties are applied on
+ * request being send to Azure Service but some could be used for other purpose also for example applicationId could be
+ * used for telemetry.
+ */
+@Fluent
+public final class ClientOptions {
+    private final Map<String, Header> headers = new ConcurrentHashMap<>();
+    private String applicationId;
+
+    /**
+     * Gets the applicationId.
+     *
+     * @return The applicationId.
+     */
+    public String getApplicationId() {
+        return applicationId;
+    }
+
+    /**
+     * Sets the applicationId provided.
+     *
+     * @param applicationId to be set.
+     * @return updated {@link ClientOptions}.
+     *
+     * @throws NullPointerException if {@code applicationId} is null.
+     */
+    public ClientOptions setApplicationId(String applicationId) {
+        this.applicationId = Objects.requireNonNull(applicationId, "'applicationId' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Sets the provided headers.
+     *
+     * @param headers headers to be set.
+     * @return updated {@link ClientOptions}.
+     *
+     * @throws NullPointerException if {@code headers} is null.
+     */
+    public ClientOptions headers(Iterable<Header> headers) {
+        Objects.requireNonNull(headers, "'headers' cannot be null.");
+
+        for (final Header header : headers) {
+            this.headers.put(formatKey(header.getName()), header);
+        }
+        return this;
+    }
+
+    /**
+     * Sets a {@link Header header} with the given name and value.
+     *
+     * <p>If header with same name already exists then the value will be overwritten.</p>
```

Comment:
This is not consistent with the behavior of Header itself, where the value gets appended to the header.